--- a/src/PYTHON/pair_python.cpp
+++ b/src/PYTHON/pair_python.cpp
@@ -53,7 +53,7 @@
   PyList_Append(py_path, PY_STRING_FROM_STRING("."));
 
   // if LAMMPS_POTENTIALS environment variable is set, add it to PYTHONPATH as well
-  const char * potentials_path = getenv("LAMMPS_POTENTIALS");
+  const char * potentials_path = "/usr/share/lammps/potentials";
   if (potentials_path != NULL) {
     PyList_Append(py_path, PY_STRING_FROM_STRING(potentials_path));
   }
--- a/src/force.cpp
+++ b/src/force.cpp
@@ -1055,7 +1055,7 @@
 
   // try the environment variable directory
 
-  const char *path = getenv("LAMMPS_POTENTIALS");
+  const char *path = "/usr/share/lammps/potentials";
   if (path == NULL) return NULL;
 
   const char *pot = potential_name(name);
