diff -crN xtapp.orig/src/Makefile xtapp/src/Makefile
*** xtapp.orig/src/Makefile	Mon Jan 19 15:37:56 2015
--- xtapp/src/Makefile	Mon Jan 19 15:38:42 2015
***************
*** 52,67 ****
  #--------------------------------
  # for Intel Linux and Intel Fortran Compiler 11.0
  #
! #MDOBJ = mdunistd.o paraenv.o
  #
  # -DINCONSISTENT_THREAD_COMP
  #    when thread computing is inconsistent, define this
! #DEFINES = ${GDEFS} -DINCONSISTENT_THREAD_COMP
  
! #FFLAGS = -O3 -openmp -msse3 -fno-inline-functions -traceback ${DEFINES}
! #F77FLAGS =
! #F90FLAGS =
! #FOPTNON = -O0
  
  # for NVIDIA CUDA
  #LDFLAGS = -openmp -L /usr/local/cuda40/cuda/lib64 -lcutil_x86_64 -lcufft -lcuda
--- 52,67 ----
  #--------------------------------
  # for Intel Linux and Intel Fortran Compiler 11.0
  #
! MDOBJ = mdunistd.o paraenv.o
  #
  # -DINCONSISTENT_THREAD_COMP
  #    when thread computing is inconsistent, define this
! DEFINES = ${GDEFS} -DINCONSISTENT_THREAD_COMP
  
! FFLAGS = -O3 -openmp -msse3 -fno-inline-functions -traceback ${DEFINES}
! F77FLAGS =
! F90FLAGS =
! FOPTNON = -O0
  
  # for NVIDIA CUDA
  #LDFLAGS = -openmp -L /usr/local/cuda40/cuda/lib64 -lcutil_x86_64 -lcufft -lcuda
***************
*** 78,86 ****
  #CFLAGS = -openmp -O3 -fno-inline-functions
  #
  # for multi-thread
! #LIBLAPACK = -lmkl_lapack95_lp64 # for composerxe-2011
  #LIBLAPACK = -lmkl_lapack
! #LIBBLAS = -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core -liomp5 -lpthread
  #
  # for single-thread
  ##LIBLAPACK = -lmkl_lapack95_lp64 # for composerxe-2011
--- 78,86 ----
  #CFLAGS = -openmp -O3 -fno-inline-functions
  #
  # for multi-thread
! LIBLAPACK = -lmkl_lapack95_lp64 # for composerxe-2011
  #LIBLAPACK = -lmkl_lapack
! LIBBLAS = -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core -liomp5 -lpthread
  #
  # for single-thread
  ##LIBLAPACK = -lmkl_lapack95_lp64 # for composerxe-2011
***************
*** 100,121 ****
  #	-lgdrdgemm-ovl -L $(GDRBASEDIR)/gdrtb3/singlibMC_tb4 \
  #	-lsing -L $(GDRBASEDIR)/gdrtb3/hibutil -lhib
  
! #ARCH_OBJ = arch_dgemm.o
  #ARCH_OBJ = arch_gdr.o
  #ARCH_OBJ = arch_nehalem.o
  #arch_nehalem.o: arch_nehalem.f90 timeprof.o
  #	${FC} -O3 -openmp -msse3 -fno-inline-functions -unroll=0 -traceback ${DEFINES} -c $<
  
! #XMPI_OBJ = xmpi_univ.o
  
! #EIGSYSTM_OBJ = eigsystm-lapack.o
  #EIGSYSTM_OBJ = eigsystm-scalapack.o
  
! #FFT3D_OBJ = fft3d_mkl.o
! #mkl_dfti.o: mkl_dfti.f90
! #mkl_dfti.f90:
! #	cp ${MKLROOT}/include/mkl_dfti.f90 .
! #FFTGRID_H_SRC = fftgrid-scl.h
  
  #FFT3D_OBJ = fft3d_fftw.o
  #FFTGRID_H_SRC = fftgrid-scl-fftw.h
--- 100,121 ----
  #	-lgdrdgemm-ovl -L $(GDRBASEDIR)/gdrtb3/singlibMC_tb4 \
  #	-lsing -L $(GDRBASEDIR)/gdrtb3/hibutil -lhib
  
! ARCH_OBJ = arch_dgemm.o
  #ARCH_OBJ = arch_gdr.o
  #ARCH_OBJ = arch_nehalem.o
  #arch_nehalem.o: arch_nehalem.f90 timeprof.o
  #	${FC} -O3 -openmp -msse3 -fno-inline-functions -unroll=0 -traceback ${DEFINES} -c $<
  
! XMPI_OBJ = xmpi_univ.o
  
! EIGSYSTM_OBJ = eigsystm-lapack.o
  #EIGSYSTM_OBJ = eigsystm-scalapack.o
  
! FFT3D_OBJ = fft3d_mkl.o
! mkl_dfti.o: mkl_dfti.f90
! mkl_dfti.f90:
! 	cp ${MKLROOT}/include/mkl_dfti.f90 .
! FFTGRID_H_SRC = fftgrid-scl.h
  
  #FFT3D_OBJ = fft3d_fftw.o
  #FFTGRID_H_SRC = fftgrid-scl-fftw.h
***************
*** 167,198 ****
  #--------------------------------
  #for x86_64 Debian Linux by S. Todo (2014.3)
  #
! ARCH_OBJ = arch_dgemm.o giounit_open.o giounit_util.o
! MDOBJ = mdunistd.o paraenv.o
! DEFINES = -O3 ${GDEFS} -DINCONSISTENT_THREAD_COMP
  
! FFLAGS = -O3 -fopenmp -msse3 -fno-inline-functions ${DEFINES} -I/usr/include
! F77FLAGS =
! F90FLAGS = -ffree-line-length-none
! FOPTNON = -O0
  
! LDFLAGS = -fopenmp
! CFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
! CXXFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
! 
! LIBLAPACK = -lfftw3_threads -llapack
! LIBBLAS = -lblas
! EXTRALIBS = -lstdc++
  
! XMPI_OBJ = xmpi_univ.o
! EIGSYSTM_OBJ = eigsystm-lapack.o
! FFT3D_OBJ = fft3d_fftw.o
! FFTGRID_H_SRC = fftgrid-scl-fftw.h
  
! CC = gcc
! CXX = g++
! FC_CIF = gfortran
! FFLAG_CIF = 
  #--------------------------------
  # for K computer
  #
--- 167,198 ----
  #--------------------------------
  #for x86_64 Debian Linux by S. Todo (2014.3)
  #
! #ARCH_OBJ = arch_dgemm.o giounit_open.o giounit_util.o
! #MDOBJ = mdunistd.o paraenv.o
! #DEFINES = -O3 ${GDEFS} -DINCONSISTENT_THREAD_COMP
  
! #FFLAGS = -O3 -fopenmp -msse3 -fno-inline-functions ${DEFINES} -I/usr/include
! #F77FLAGS =
! #F90FLAGS = -ffree-line-length-none
! #FOPTNON = -O0
  
! #LDFLAGS = -fopenmp
! #CFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
! #CXXFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
  
! #LIBLAPACK = -lfftw3_threads -llapack
! #LIBBLAS = -lblas
! #EXTRALIBS = -lstdc++
! 
! #XMPI_OBJ = xmpi_univ.o
! #EIGSYSTM_OBJ = eigsystm-lapack.o
! #FFT3D_OBJ = fft3d_fftw.o
! #FFTGRID_H_SRC = fftgrid-scl-fftw.h
  
! #CC = gcc
! #CXX = g++
! #FC_CIF = gfortran
! #FFLAG_CIF = 
  #--------------------------------
  # for K computer
  #
***************
*** 535,542 ****
  	${CPP} -P ${DEFINES} $< | sed '/^ *$$/d' > $*.f
  
  install: default
! 	install -d $(DESTDIR)/usr/bin
! 	install inipot cgmrpt mdrpt vbpef vbstm wfn2chg wannier pefcos xticonv strconv diffstr iplstr cmpstr pe2dos tetrapdos hmatr2bnd $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 535,542 ----
  	${CPP} -P ${DEFINES} $< | sed '/^ *$$/d' > $*.f
  
  install: default
! 	mkdir -p $(PREFIX)/bin
! 	install inipot cgmrpt mdrpt vbpef vbstm wfn2chg wannier pefcos xticonv strconv diffstr iplstr cmpstr pe2dos tetrapdos hmatr2bnd $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/tapioca-1.4.2/src/Makefile xtapp/tapioca-1.4.2/src/Makefile
*** xtapp.orig/tapioca-1.4.2/src/Makefile	Mon Jan 19 15:37:59 2015
--- xtapp/tapioca-1.4.2/src/Makefile	Mon Jan 19 15:39:49 2015
***************
*** 71,77 ****
  test: $(TARGET)
  
  install: $(TARGET)
! 	install -d $(DESTDIR)/usr/libexec/tapioca
! 	install $(TARGET) $(DESTDIR)/usr/libexec/tapioca
  
  include Makefile.dep
--- 71,77 ----
  test: $(TARGET)
  
  install: $(TARGET)
! 	mkdir -p $(PREFIX)/libexec
! 	install $(TARGET) $(PREFIX)/libexec
  
  include Makefile.dep
diff -crN xtapp.orig/tapioca-1.4.2/src/Makefile.inc xtapp/tapioca-1.4.2/src/Makefile.inc
*** xtapp.orig/tapioca-1.4.2/src/Makefile.inc	Mon Jan 19 15:37:59 2015
--- xtapp/tapioca-1.4.2/src/Makefile.inc	Mon Jan 19 15:40:57 2015
***************
*** 7,27 ****
  #ICON   =  icon.o
  
  ### for Linux, SuSE
! #QT4    =  /usr/lib64/qt4
! #CC     =  g++
! #CFLAGS = -O3 -Wall
! #INCS   = -I$(QT4)/include -I/usr/X11R6/include
! #LIBS   = -L$(QT4)/lib64 -lQtCore -lQtGui -lQtOpenGL -lQtXml -llapack -lblas
! #ICON   = 
! 
! ### for Linux, debian 6
! QT4    =  /usr
  CC     =  g++
  CFLAGS = -O3 -Wall
! INCS   = -I/usr/include/qt4 -I/usr/X11R6/include
! LIBS   = -lQtCore -lQtGui -lQtOpenGL -lQtXml -lGLU -llapack -lblas
  ICON   = 
  
  ### for Mac
  #QT4    =  /usr/local/Trolltech/Qt-4.8.4
  #CC     =  g++
--- 7,27 ----
  #ICON   =  icon.o
  
  ### for Linux, SuSE
! QT4    =  /usr/lib64/qt4
  CC     =  g++
  CFLAGS = -O3 -Wall
! INCS   = -I$(QT4)/include -I/usr/X11R6/include
! LIBS   = -L$(QT4)/lib64 -lQtCore -lQtGui -lQtOpenGL -lQtXml -llapack -lblas
  ICON   = 
  
+ ### for Linux, debian 6
+ #QT4    =  /usr
+ #CC     =  g++
+ #CFLAGS = -O3 -Wall
+ #INCS   = -I/usr/include/qt4 -I/usr/X11R6/include
+ #LIBS   = -lQtCore -lQtGui -lQtOpenGL -lQtXml -lGLU -llapack -lblas
+ #ICON   = 
+ 
  ### for Mac
  #QT4    =  /usr/local/Trolltech/Qt-4.8.4
  #CC     =  g++
diff -crN xtapp.orig/xtapp-util/fldtool/Makefile xtapp/xtapp-util/fldtool/Makefile
*** xtapp.orig/xtapp-util/fldtool/Makefile	Mon Jan 19 15:37:56 2015
--- xtapp/xtapp-util/fldtool/Makefile	Mon Jan 19 15:38:42 2015
***************
*** 64,71 ****
  	fftgrid.h
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install wfn2dx lpt2dx rho2dx rok2dx wfk2dx stm2dx rhomm lptmm wfn2ee $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 64,71 ----
  	fftgrid.h
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	install wfn2dx lpt2dx rho2dx rok2dx wfk2dx stm2dx rhomm lptmm wfn2ee $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/xtapp-util/inputconv/Makefile xtapp/xtapp-util/inputconv/Makefile
*** xtapp.orig/xtapp-util/inputconv/Makefile	Mon Jan 19 15:37:56 2015
--- xtapp/xtapp-util/inputconv/Makefile	Mon Jan 19 15:38:42 2015
***************
*** 39,46 ****
  	inputconv a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install inputconv $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 39,46 ----
  	inputconv a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	install inputconv $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/xtapp-util/mdtool/Makefile xtapp/xtapp-util/mdtool/Makefile
*** xtapp.orig/xtapp-util/mdtool/Makefile	Mon Jan 19 15:37:56 2015
--- xtapp/xtapp-util/mdtool/Makefile	Mon Jan 19 15:38:42 2015
***************
*** 40,47 ****
  	scanmdlog tailmdlog a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install scanmdlog tailmdlog $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 40,47 ----
  	scanmdlog tailmdlog a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	install scanmdlog tailmdlog $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/xtapp-util/minmaxband/Makefile xtapp/xtapp-util/minmaxband/Makefile
*** xtapp.orig/xtapp-util/minmaxband/Makefile	Mon Jan 19 15:37:56 2015
--- xtapp/xtapp-util/minmaxband/Makefile	Mon Jan 19 15:38:42 2015
***************
*** 22,29 ****
  	minmaxband a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install minmaxband $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 22,29 ----
  	minmaxband a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	install minmaxband $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/xtapp-util/wfchg2pdos/Makefile xtapp/xtapp-util/wfchg2pdos/Makefile
*** xtapp.orig/xtapp-util/wfchg2pdos/Makefile	Mon Jan 19 15:37:56 2015
--- xtapp/xtapp-util/wfchg2pdos/Makefile	Mon Jan 19 15:38:42 2015
***************
*** 40,47 ****
  	ltzpdos wfchg2pdos a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install ltzpdos wfchg2pdos $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 40,47 ----
  	ltzpdos wfchg2pdos a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	install ltzpdos wfchg2pdos $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
