diff -crN xtapp_150401.orig/src/Makefile xtapp_150401/src/Makefile
*** xtapp_150401.orig/src/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp_150401/src/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 171,186 ****
  MDOBJ = mdunistd.o paraenv.o
  DEFINES = -O3 ${GDEFS} -DINCONSISTENT_THREAD_COMP
  
! FFLAGS = -O3 -fopenmp -msse3 -fno-inline-functions ${DEFINES} -I/usr/include
  F77FLAGS =
  F90FLAGS = -ffree-line-length-none
  FOPTNON = -O0
  
  LDFLAGS = -fopenmp
! CFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
! CXXFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
  
! LIBLAPACK = -lfftw3_threads -llapack
  LIBBLAS = -lblas
  EXTRALIBS = -lstdc++
  
--- 171,186 ----
  MDOBJ = mdunistd.o paraenv.o
  DEFINES = -O3 ${GDEFS} -DINCONSISTENT_THREAD_COMP
  
! FFLAGS = -O3 -fopenmp -msse3 -fno-inline-functions ${DEFINES} -I/opt/local/include
  F77FLAGS =
  F90FLAGS = -ffree-line-length-none
  FOPTNON = -O0
  
  LDFLAGS = -fopenmp
! CFLAGS = -fopenmp -O3 -fno-inline-functions -I/opt/local/include
! CXXFLAGS = -fopenmp -O3 -fno-inline-functions -I/opt/local/include
  
! LIBLAPACK = -L/opt/local/lib -lfftw3_threads -llapack
  LIBBLAS = -lblas
  EXTRALIBS = -lstdc++
  
***************
*** 535,542 ****
  	${CPP} -P ${DEFINES} $< | sed '/^ *$$/d' > $*.f
  
  install: default
! 	install -d $(DESTDIR)/usr/bin
! 	install inipot cgmrpt mdrpt vbpef vbstm wfn2chg wannier pefcos xticonv strconv diffstr iplstr cmpstr pe2dos tetrapdos hmatr2bnd $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 535,542 ----
  	${CPP} -P ${DEFINES} $< | sed '/^ *$$/d' > $*.f
  
  install: default
! 	mkdir -p $(PREFIX)/bin
! 	cp -p inipot cgmrpt mdrpt vbpef vbstm wfn2chg wannier pefcos xticonv strconv diffstr iplstr cmpstr pe2dos tetrapdos hmatr2bnd $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
*** xtapp_150401.orig/xtapp-util_150401/fldtool/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp_150401/xtapp-util_150401/fldtool/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 64,71 ****
  	fftgrid.h
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install wfn2dx lpt2dx rho2dx rok2dx wfk2dx stm2dx rhomm lptmm wfn2ee $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 64,71 ----
  	fftgrid.h
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p wfn2dx lpt2dx rho2dx rok2dx wfk2dx stm2dx rhomm lptmm wfn2ee $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp_150401.orig/xtapp-util_150401/inputconv/Makefile xtapp_150401/xtapp-util_150401/inputconv/Makefile
*** xtapp_150401.orig/xtapp-util_150401/inputconv/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp_150401/xtapp-util_150401/inputconv/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 39,46 ****
  	inputconv a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install inputconv $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 39,46 ----
  	inputconv a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p inputconv $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp_150401.orig/xtapp-util_150401/mdtool/Makefile xtapp_150401/xtapp-util_150401/mdtool/Makefile
*** xtapp_150401.orig/xtapp-util_150401/mdtool/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp_150401/xtapp-util_150401/mdtool/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 40,47 ****
  	scanmdlog tailmdlog a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install scanmdlog tailmdlog $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 40,47 ----
  	scanmdlog tailmdlog a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p scanmdlog tailmdlog $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp_150401.orig/xtapp-util_150401/minmaxband/Makefile xtapp_150401/xtapp-util_150401/minmaxband/Makefile
*** xtapp_150401.orig/xtapp-util_150401/minmaxband/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp_150401/xtapp-util_150401/minmaxband/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 22,29 ****
  	minmaxband a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install minmaxband $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 22,29 ----
  	minmaxband a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p minmaxband $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp_150401.orig/xtapp-util_150401/wfchg2pdos/Makefile xtapp_150401/xtapp-util_150401/wfchg2pdos/Makefile
*** xtapp_150401.orig/xtapp-util_150401/wfchg2pdos/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp_150401/xtapp-util_150401/wfchg2pdos/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 40,47 ****
  	ltzpdos wfchg2pdos a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install ltzpdos wfchg2pdos $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 40,47 ----
  	ltzpdos wfchg2pdos a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p ltzpdos wfchg2pdos $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
