diff -crN xtapp.orig/src/Makefile xtapp/src/Makefile
*** xtapp.orig/src/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp/src/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 171,186 ****
  MDOBJ = mdunistd.o paraenv.o
  DEFINES = -O3 ${GDEFS} -DINCONSISTENT_THREAD_COMP
  
! FFLAGS = -O3 -fopenmp -msse3 -fno-inline-functions ${DEFINES} -I/usr/include
  F77FLAGS =
  F90FLAGS = -ffree-line-length-none
  FOPTNON = -O0
  
  LDFLAGS = -fopenmp
! CFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
! CXXFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
  
! LIBLAPACK = -lfftw3_threads -llapack
  LIBBLAS = -lblas
  EXTRALIBS = -lstdc++
  
--- 171,186 ----
  MDOBJ = mdunistd.o paraenv.o
  DEFINES = -O3 ${GDEFS} -DINCONSISTENT_THREAD_COMP
  
! FFLAGS = -O3 -fopenmp -msse3 -fno-inline-functions ${DEFINES} -I/opt/local/include
  F77FLAGS =
  F90FLAGS = -ffree-line-length-none
  FOPTNON = -O0
  
  LDFLAGS = -fopenmp
! CFLAGS = -fopenmp -O3 -fno-inline-functions -I/opt/local/include
! CXXFLAGS = -fopenmp -O3 -fno-inline-functions -I/opt/local/include
  
! LIBLAPACK = -L/opt/local/lib -lfftw3_threads -llapack
  LIBBLAS = -lblas
  EXTRALIBS = -lstdc++
  
***************
*** 535,542 ****
  	${CPP} -P ${DEFINES} $< | sed '/^ *$$/d' > $*.f
  
  install: default
! 	install -d $(DESTDIR)/usr/bin
! 	install inipot cgmrpt mdrpt vbpef vbstm wfn2chg wannier pefcos xticonv strconv diffstr iplstr cmpstr pe2dos tetrapdos hmatr2bnd $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 535,542 ----
  	${CPP} -P ${DEFINES} $< | sed '/^ *$$/d' > $*.f
  
  install: default
! 	mkdir -p $(PREFIX)/bin
! 	cp -p inipot cgmrpt mdrpt vbpef vbstm wfn2chg wannier pefcos xticonv strconv diffstr iplstr cmpstr pe2dos tetrapdos hmatr2bnd $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/tapioca-1.4.2/src/Makefile xtapp/tapioca-1.4.2/src/Makefile
*** xtapp.orig/tapioca-1.4.2/src/Makefile	Mon Jan 19 14:41:33 2015
--- xtapp/tapioca-1.4.2/src/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 71,77 ****
  test: $(TARGET)
  
  install: $(TARGET)
! 	install -d $(DESTDIR)/usr/libexec/tapioca
! 	install $(TARGET) $(DESTDIR)/usr/libexec/tapioca
  
  include Makefile.dep
--- 71,77 ----
  test: $(TARGET)
  
  install: $(TARGET)
! 	mkdir -p $(PREFIX)/libexec
! 	cp -p $(TARGET) $(PREFIX)/libexec
  
  include Makefile.dep
diff -crN xtapp.orig/tapioca-1.4.2/src/Makefile.inc xtapp/tapioca-1.4.2/src/Makefile.inc
*** xtapp.orig/tapioca-1.4.2/src/Makefile.inc	Mon Jan 19 14:41:33 2015
--- xtapp/tapioca-1.4.2/src/Makefile.inc	Mon Jan 19 14:44:05 2015
***************
*** 15,35 ****
  #ICON   = 
  
  ### for Linux, debian 6
! QT4    =  /usr
! CC     =  g++
! CFLAGS = -O3 -Wall
! INCS   = -I/usr/include/qt4 -I/usr/X11R6/include
! LIBS   = -lQtCore -lQtGui -lQtOpenGL -lQtXml -lGLU -llapack -lblas
! ICON   = 
! 
! ### for Mac
! #QT4    =  /usr/local/Trolltech/Qt-4.8.4
  #CC     =  g++
  #CFLAGS = -O3 -Wall
! #INCS   = -I$(QT4)/include -I/usr/X11R6/include
! #LIBS   = -F$(QT4)/lib -L$(QT4)/lib -framework QtXml -framework QtOpenGL -framework QtGui -framework QtCore -framework OpenGL -llapack -lblas
  #ICON   = 
  
  #########################################
  ### do not edit lines below
  MOC    = $(QT4)/bin/moc
--- 15,35 ----
  #ICON   = 
  
  ### for Linux, debian 6
! #QT4    =  /usr
  #CC     =  g++
  #CFLAGS = -O3 -Wall
! #INCS   = -I/usr/include/qt4 -I/usr/X11R6/include
! #LIBS   = -lQtCore -lQtGui -lQtOpenGL -lQtXml -lGLU -llapack -lblas
  #ICON   = 
  
+ ### for Mac
+ QT4    =  /opt/local
+ CC     =  g++
+ CFLAGS = -O3 -Wall
+ INCS   = -I$(QT4)/include
+ LIBS   = -F$(QT4)/Library/Frameworks -L$(QT4)/lib -framework QtXml -framework QtOpenGL -framework QtGui -framework QtCore -framework OpenGL -llapack -lblas
+ ICON   = 
+ 
  #########################################
  ### do not edit lines below
  MOC    = $(QT4)/bin/moc
diff -crN xtapp.orig/xtapp-util/fldtool/Makefile xtapp/xtapp-util/fldtool/Makefile
*** xtapp.orig/xtapp-util/fldtool/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp/xtapp-util/fldtool/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 64,71 ****
  	fftgrid.h
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install wfn2dx lpt2dx rho2dx rok2dx wfk2dx stm2dx rhomm lptmm wfn2ee $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 64,71 ----
  	fftgrid.h
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p wfn2dx lpt2dx rho2dx rok2dx wfk2dx stm2dx rhomm lptmm wfn2ee $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/xtapp-util/inputconv/Makefile xtapp/xtapp-util/inputconv/Makefile
*** xtapp.orig/xtapp-util/inputconv/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp/xtapp-util/inputconv/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 39,46 ****
  	inputconv a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install inputconv $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 39,46 ----
  	inputconv a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p inputconv $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/xtapp-util/mdtool/Makefile xtapp/xtapp-util/mdtool/Makefile
*** xtapp.orig/xtapp-util/mdtool/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp/xtapp-util/mdtool/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 40,47 ****
  	scanmdlog tailmdlog a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install scanmdlog tailmdlog $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 40,47 ----
  	scanmdlog tailmdlog a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p scanmdlog tailmdlog $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/xtapp-util/minmaxband/Makefile xtapp/xtapp-util/minmaxband/Makefile
*** xtapp.orig/xtapp-util/minmaxband/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp/xtapp-util/minmaxband/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 22,29 ****
  	minmaxband a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install minmaxband $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 22,29 ----
  	minmaxband a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p minmaxband $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp.orig/xtapp-util/wfchg2pdos/Makefile xtapp/xtapp-util/wfchg2pdos/Makefile
*** xtapp.orig/xtapp-util/wfchg2pdos/Makefile	Mon Jan 19 14:41:30 2015
--- xtapp/xtapp-util/wfchg2pdos/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 40,47 ****
  	ltzpdos wfchg2pdos a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install ltzpdos wfchg2pdos $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 40,47 ----
  	ltzpdos wfchg2pdos a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p ltzpdos wfchg2pdos $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
