diff -urN cmake-2.8.12.1.orig/Modules/CMakeParseImplicitLinkInfo.cmake cmake-2.8.12.1/Modules/CMakeParseImplicitLinkInfo.cmake
--- cmake-2.8.12.1.orig/Modules/CMakeParseImplicitLinkInfo.cmake	2013-11-06 04:07:22.000000000 +0900
+++ cmake-2.8.12.1/Modules/CMakeParseImplicitLinkInfo.cmake	2013-11-23 01:47:33.000000000 +0900
@@ -26,11 +26,10 @@
   set(linker "CMAKE_LINKER-NOTFOUND")
   if(CMAKE_LINKER)
     get_filename_component(linker ${CMAKE_LINKER} NAME)
-    string(REGEX REPLACE "([][+.*?()^$])" "\\\\\\1" linker "${linker}")
   endif()
   # Construct a regex to match linker lines.  It must match both the
   # whole line and just the command (argv[0]).
-  set(linker_regex "^( *|.*[/\\])(${linker}|([^/\\]+-)?ld|collect2)[^/\\]*( |$)")
+  set(linker_regex "^( *|.*[/\\])(${linker}|ld|collect2)[^/\\]*( |$)")
   set(linker_exclude_regex "collect2 version ")
   set(log "${log}  link line regex: [${linker_regex}]\n")
   string(REGEX REPLACE "\r?\n" ";" output_lines "${text}")
