diff -crN alps_20160816~r7711.orig/config/FindBoostForALPS.cmake alps_20160816~r7711/config/FindBoostForALPS.cmake
*** alps_20160816~r7711.orig/config/FindBoostForALPS.cmake	2013-10-16 20:37:49.000000000 +0900
--- alps_20160816~r7711/config/FindBoostForALPS.cmake	2016-09-20 10:30:05.000000000 +0900
***************
*** 27,40 ****
    # before actual find_package for Boost.
    # - Ubuntu 10.10 does not have Boost.MPI package.
    set(MPI)
!   find_package(Boost 1.48.0 COMPONENTS mpi)
    if (Boost_FOUND)
      set(MPI mpi)
    endif (Boost_FOUND)
    # Linking to Boost.test Unit Test Framework is optional
    set(UTF)
    if (ALPS_LINK_BOOST_TEST)
!     find_package(Boost 1.48.0 COMPONENTS unit_test_framework)
      if (Boost_FOUND)
        set(UTF unit_test_framework)
        # Unset ALPS_INSTALL_BOOST_TEST
--- 27,40 ----
    # before actual find_package for Boost.
    # - Ubuntu 10.10 does not have Boost.MPI package.
    set(MPI)
!   find_package(Boost 1.52.0 COMPONENTS mpi)
    if (Boost_FOUND)
      set(MPI mpi)
    endif (Boost_FOUND)
    # Linking to Boost.test Unit Test Framework is optional
    set(UTF)
    if (ALPS_LINK_BOOST_TEST)
!     find_package(Boost 1.52.0 COMPONENTS unit_test_framework)
      if (Boost_FOUND)
        set(UTF unit_test_framework)
        # Unset ALPS_INSTALL_BOOST_TEST
***************
*** 49,60 ****
    # However, Boost.timer (v2) is built and linked when boost is built together.
    # Therefore, When Boost.timer (v2) is found, link ALPS against it.
    set(TIMER)
!   find_package(Boost 1.48.0 COMPONENTS timer)
    if (Boost_FOUND)
      set(TIMER timer)
    endif (Boost_FOUND)
    # The final library finding for precompiled Boost.
!   find_package(Boost 1.48.0 COMPONENTS chrono date_time filesystem iostreams program_options python regex system serialization thread ${MPI} ${UTF} ${TIMER})
    # Unset local variable
    unset(MPI)
    unset(UTF)
--- 49,64 ----
    # However, Boost.timer (v2) is built and linked when boost is built together.
    # Therefore, When Boost.timer (v2) is found, link ALPS against it.
    set(TIMER)
!   find_package(Boost 1.52.0 COMPONENTS timer)
    if (Boost_FOUND)
      set(TIMER timer)
    endif (Boost_FOUND)
    # The final library finding for precompiled Boost.
!   if(ALPS_BUILD_PYTHON)
!     find_package(Boost 1.52.0 COMPONENTS chrono date_time filesystem iostreams program_options python regex system serialization thread ${MPI} ${UTF} ${TIMER})
!   else(ALPS_BUILD_PYTHON)
!     find_package(Boost 1.52.0 COMPONENTS chrono date_time filesystem iostreams program_options regex system serialization thread ${MPI} ${UTF} ${TIMER})
!   endif(ALPS_BUILD_PYTHON)
    # Unset local variable
    unset(MPI)
    unset(UTF)
