Index: alps_20180220~r7849/config/FindBoostForALPS.cmake
===================================================================
--- alps_20180220~r7849.orig/config/FindBoostForALPS.cmake	2018-02-20 10:07:41.000000000 +0900
+++ alps_20180220~r7849/config/FindBoostForALPS.cmake	2018-06-05 13:35:02.000000000 +0900
@@ -21,20 +21,20 @@
   # Debug flag for FindBoost.cmake
   #  set(Boost_DEBUG TRUE)
 
-  message(STATUS "Looking for precompiled Boost libraries (version >= 1.52)")
+  message(STATUS "Looking for precompiled Boost libraries (version >= 1.49)")
 
   # We do not require Boost.MPI, therefore check whether Boost.MPI exists
   # before actual find_package for Boost.
   # - Ubuntu 10.10 does not have Boost.MPI package.
   set(MPI)
-  find_package(Boost 1.52.0 COMPONENTS mpi)
+  find_package(Boost 1.49.0 COMPONENTS mpi)
   if (Boost_FOUND)
     set(MPI mpi)
   endif (Boost_FOUND)
   # Linking to Boost.test Unit Test Framework is optional
   set(UTF)
   if (ALPS_LINK_BOOST_TEST)
-    find_package(Boost 1.52.0 COMPONENTS unit_test_framework)
+    find_package(Boost 1.49.0 COMPONENTS unit_test_framework)
     if (Boost_FOUND)
       set(UTF unit_test_framework)
       # Unset ALPS_INSTALL_BOOST_TEST
@@ -62,9 +62,9 @@
   endif (Boost_FOUND)
   # The final library finding for precompiled Boost.
   if(ALPS_BUILD_PYTHON)
-    find_package(Boost 1.52.0 COMPONENTS chrono date_time filesystem iostreams program_options python regex system serialization thread ${MPI} ${UTF} ${TIMER} ${NUMPY})
+    find_package(Boost 1.49.0 COMPONENTS chrono date_time filesystem iostreams program_options python regex system serialization thread ${MPI} ${UTF} ${TIMER} ${NUMPY})
   else(ALPS_BUILD_PYTHON)
-    find_package(Boost 1.52.0 COMPONENTS chrono date_time filesystem iostreams program_options regex system serialization thread ${MPI} ${UTF} ${TIMER} ${NUMPY})
+    find_package(Boost 1.49.0 COMPONENTS chrono date_time filesystem iostreams program_options regex system serialization thread ${MPI} ${UTF} ${TIMER} ${NUMPY})
   endif(ALPS_BUILD_PYTHON)
   # Unset local variable
   unset(MPI)
