diff -crN alps-20140207-r7338.orig/applications/dmrg/CMakeLists.txt alps-20140207-r7338/applications/dmrg/CMakeLists.txt
*** alps-20140207-r7338.orig/applications/dmrg/CMakeLists.txt	Fri Jan 10 00:13:16 2014
--- alps-20140207-r7338/applications/dmrg/CMakeLists.txt	Sun Feb  9 22:39:02 2014
***************
*** 3,8 ****
  #      (See accompanying file LICENSE_1_0.txt or copy at
  #          http://www.boost.org/LICENSE_1_0.txt)
  
! add_subdirectory(mps)
  add_subdirectory(dmrg)
  add_subdirectory(tebd)
--- 3,8 ----
  #      (See accompanying file LICENSE_1_0.txt or copy at
  #          http://www.boost.org/LICENSE_1_0.txt)
  
! # add_subdirectory(mps)
  add_subdirectory(dmrg)
  add_subdirectory(tebd)
