diff -crN tapioca-1.4.2.orig/src/Makefile tapioca-1.4.2/src/Makefile
*** tapioca-1.4.2.orig/src/Makefile	Mon Jan 19 14:41:33 2015
--- tapioca-1.4.2/src/Makefile	Mon Jan 19 14:42:21 2015
***************
*** 71,77 ****
  test: $(TARGET)
  
  install: $(TARGET)
! 	install -d $(DESTDIR)/usr/libexec/tapioca
! 	install $(TARGET) $(DESTDIR)/usr/libexec/tapioca
  
  include Makefile.dep
--- 71,77 ----
  test: $(TARGET)
  
  install: $(TARGET)
! 	mkdir -p $(PREFIX)/libexec
! 	cp -p $(TARGET) $(PREFIX)/libexec
  
  include Makefile.dep
diff -crN tapioca-1.4.2.orig/src/Makefile.inc tapioca-1.4.2/src/Makefile.inc
*** tapioca-1.4.2.orig/src/Makefile.inc	Mon Jan 19 14:41:33 2015
--- tapioca-1.4.2/src/Makefile.inc	Mon Jan 19 14:44:05 2015
***************
*** 15,35 ****
  #ICON   = 
  
  ### for Linux, debian 6
! QT4    =  /usr
! CC     =  g++
! CFLAGS = -O3 -Wall
! INCS   = -I/usr/include/qt4 -I/usr/X11R6/include
! LIBS   = -lQtCore -lQtGui -lQtOpenGL -lQtXml -lGLU -llapack -lblas
! ICON   = 
! 
! ### for Mac
! #QT4    =  /usr/local/Trolltech/Qt-4.8.4
  #CC     =  g++
  #CFLAGS = -O3 -Wall
! #INCS   = -I$(QT4)/include -I/usr/X11R6/include
! #LIBS   = -F$(QT4)/lib -L$(QT4)/lib -framework QtXml -framework QtOpenGL -framework QtGui -framework QtCore -framework OpenGL -llapack -lblas
  #ICON   = 
  
  #########################################
  ### do not edit lines below
  MOC    = $(QT4)/bin/moc
--- 15,35 ----
  #ICON   = 
  
  ### for Linux, debian 6
! #QT4    =  /usr
  #CC     =  g++
  #CFLAGS = -O3 -Wall
! #INCS   = -I/usr/include/qt4 -I/usr/X11R6/include
! #LIBS   = -lQtCore -lQtGui -lQtOpenGL -lQtXml -lGLU -llapack -lblas
  #ICON   = 
  
+ ### for Mac
+ QT4    =  /opt/local
+ CC     =  g++
+ CFLAGS = -O3 -Wall
+ INCS   = -I$(QT4)/include
+ LIBS   = -F$(QT4)/Library/Frameworks -L$(QT4)/lib -framework QtXml -framework QtOpenGL -framework QtGui -framework QtCore -framework OpenGL -llapack -lblas
+ ICON   = 
+ 
  #########################################
  ### do not edit lines below
  MOC    = $(QT4)/bin/moc
