diff -crN tapioca-1.4.2.orig/src/Makefile tapioca-1.4.2/src/Makefile
*** tapioca-1.4.2.orig/src/Makefile	Mon Jan 19 15:37:59 2015
--- tapioca-1.4.2/src/Makefile	Mon Jan 19 15:39:49 2015
***************
*** 71,77 ****
  test: $(TARGET)
  
  install: $(TARGET)
! 	install -d $(DESTDIR)/usr/libexec/tapioca
! 	install $(TARGET) $(DESTDIR)/usr/libexec/tapioca
  
  include Makefile.dep
--- 71,77 ----
  test: $(TARGET)
  
  install: $(TARGET)
! 	mkdir -p $(PREFIX)/libexec
! 	install $(TARGET) $(PREFIX)/libexec
  
  include Makefile.dep
diff -crN tapioca-1.4.2.orig/src/Makefile.inc tapioca-1.4.2/src/Makefile.inc
*** tapioca-1.4.2.orig/src/Makefile.inc	Mon Jan 19 15:37:59 2015
--- tapioca-1.4.2/src/Makefile.inc	Mon Jan 19 15:40:57 2015
***************
*** 7,27 ****
  #ICON   =  icon.o
  
  ### for Linux, SuSE
! #QT4    =  /usr/lib64/qt4
! #CC     =  g++
! #CFLAGS = -O3 -Wall
! #INCS   = -I$(QT4)/include -I/usr/X11R6/include
! #LIBS   = -L$(QT4)/lib64 -lQtCore -lQtGui -lQtOpenGL -lQtXml -llapack -lblas
! #ICON   = 
! 
! ### for Linux, debian 6
! QT4    =  /usr
  CC     =  g++
  CFLAGS = -O3 -Wall
! INCS   = -I/usr/include/qt4 -I/usr/X11R6/include
! LIBS   = -lQtCore -lQtGui -lQtOpenGL -lQtXml -lGLU -llapack -lblas
  ICON   = 
  
  ### for Mac
  #QT4    =  /usr/local/Trolltech/Qt-4.8.4
  #CC     =  g++
--- 7,27 ----
  #ICON   =  icon.o
  
  ### for Linux, SuSE
! QT4    =  /usr/lib64/qt4
  CC     =  g++
  CFLAGS = -O3 -Wall
! INCS   = -I/usr/include
! LIBS   = -L/usr/lib64 -lQtCore -lQtGui -lQtOpenGL -lQtXml -lGL -lGLU -llapack -lblas
  ICON   = 
  
+ ### for Linux, debian 6
+ #QT4    =  /usr
+ #CC     =  g++
+ #CFLAGS = -O3 -Wall
+ #INCS   = -I/usr/include/qt4 -I/usr/X11R6/include
+ #LIBS   = -lQtCore -lQtGui -lQtOpenGL -lQtXml -lGLU -llapack -lblas
+ #ICON   = 
+ 
  ### for Mac
  #QT4    =  /usr/local/Trolltech/Qt-4.8.4
  #CC     =  g++
