Description: adding shebangs to scripts under examples/.
Author: Andrius Merkys <andrius.merkys@gmail.com>
--- a/atomic/examples/pseudo-test/clean.sh
+++ b/atomic/examples/pseudo-test/clean.sh
@@ -1,3 +1,4 @@
+#!/bin/sh
 rm -rf results
 rm -f difference
 rm -f ld1*
--- a/atomic/examples/vdw-in-tfvw/clean.sh
+++ b/atomic/examples/vdw-in-tfvw/clean.sh
@@ -1 +1,2 @@
+#!/bin/sh
 rm -r results *.out *.dat ld1*
--- a/CPV/examples/EXX-wf-example/example-job-script
+++ b/CPV/examples/EXX-wf-example/example-job-script
@@ -13,7 +13,7 @@
 
 cd $PBS_O_WORKDIR
 
-CODE_n=<your_cp.x_location>
+CODE_n=/usr/bin/cp.x
 
 # example of number of MPI tasks = 2 * number of electronic states
 # using 256 MPI tasks and 2 OpenMP threads per MPI tasks
--- a/EPW/examples/diamond/phonons/pp.py
+++ b/EPW/examples/diamond/phonons/pp.py
@@ -1,3 +1,4 @@
+#!/usr/bin/python
 #
 # Post-processing script QE --> EPW 
 # 14/07/2015 - Samuel Ponce
--- a/EPW/examples/sic/phonons/pp.py
+++ b/EPW/examples/sic/phonons/pp.py
@@ -1,3 +1,4 @@
+#!/usr/bin/python
 #
 # Post-processing script QE --> EPW 
 # 14/07/2015 - Samuel Ponce
--- a/atomic/examples/all-electron/clean.sh
+++ b/atomic/examples/all-electron/clean.sh
@@ -1,2 +1,3 @@
+#!/bin/sh
 rm -r results
 rm difference ld1.wfc*
--- a/atomic/examples/pseudo-gen/clean.sh
+++ b/atomic/examples/pseudo-gen/clean.sh
@@ -1,3 +1,4 @@
+#!/bin/sh
 rm -rf results
 rm -f difference
 rm -f ld1*
--- a/atomic/examples/pseudo-LDA-0.5/clean.sh
+++ b/atomic/examples/pseudo-LDA-0.5/clean.sh
@@ -1,2 +1,3 @@
+#!/bin/sh
 rm -r -f results
 rm -r -f *CRASH*
--- a/EPW/examples/mgb2/phonons/pp.py
+++ b/EPW/examples/mgb2/phonons/pp.py
@@ -1,3 +1,4 @@
+#!/usr/bin/python
 #
 # Post-processing script QE --> EPW 
 # 14/07/2015 - Samuel Ponce
