--- a/install/make.inc.in
+++ b/install/make.inc.in
@@ -187,4 +187,6 @@
 WGET = @wget@
 
 # Install directory - not currently used
-PREFIX = @prefix@
+prefix = @prefix@
+exec_prefix = @exec_prefix@
+bindir = @bindir@
--- a/Makefile
+++ b/Makefile
@@ -264,11 +264,11 @@
 #########################################################
 
 install : 
-	@if test -d bin ; then mkdir -p $(PREFIX)/bin ; \
+	@if test -d bin ; then mkdir -p $(DESTDIR)$(bindir) ; \
 	for x in `find * ! -path "test-suite/*" -name *.x -type f` ; do \
-		cp $$x $(PREFIX)/bin/ ; done ; \
+		cp $$x $(DESTDIR)$(bindir)/ ; done ; \
 	fi
-	@echo 'Quantum ESPRESSO binaries installed in $(PREFIX)/bin'
+	@echo 'Quantum ESPRESSO binaries installed in $(DESTDIR)$(bindir)'
 
 #########################################################
 # Run test-suite for numerical regression testing
