diff -crN xtapp_20180420.orig/src/Makefile xtapp_20180420/src/Makefile
*** xtapp_20180420.orig/src/Makefile	Thu Feb 14 16:02:12 2019
--- xtapp_20180420/src/Makefile	Thu Feb 14 16:17:04 2019
***************
*** 133,169 ****
  #--------------------------------
  #for x86_64 Debian Linux by S. Todo (2014.3)
  #
! #ARCH_OBJ = arch_dgemm.o giounit_open.o giounit_util.o
! #MDOBJ = mdunistd.o paraenv.o
  #
  # -DINCONSISTENT_THREAD_COMP
  #    when thread computing is inconsistent, define this
! #DEFINES = -O3 ${GDEFS} -DINCONSISTENT_THREAD_COMP
! 
! #FFLAGS = -O3 -fopenmp -msse3 -fno-inline-functions ${DEFINES} -I/usr/include
! #F77FLAGS =
! #F90FLAGS = -ffree-line-length-none
! #FOPTNON = -O0
  
! #LDFLAGS = -fopenmp
! #CFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
! #CXXFLAGS = -fopenmp -O3 -fno-inline-functions -I/usr/include
! 
! #LIBLAPACK = -lfftw3_threads -lfftw3 -llapack
! #LIBBLAS = -lblas
! #EXTRALIBS = -lstdc++
! 
! #XMPI_OBJ = xmpi_univ.o
! ##EIGSYSTM_OBJ = eigsystm.o
! #EIGSYSTM_OBJ = eigsystm-lapack.o
! ##EIGSYSTM_OBJ = eigsystm-scalapack.o
! #FFT3D_OBJ = fft3d_fftw.o
! #FFTGRID_H_SRC = fftgrid-scl-fftw.h
  
! #CC = gcc
! #CXX = g++
! #FC_CIF = gfortran
! #FFLAG_CIF = 
  #--------------------------------
  # for K computer
  #
--- 133,169 ----
  #--------------------------------
  #for x86_64 Debian Linux by S. Todo (2014.3)
  #
! ARCH_OBJ = arch_dgemm.o giounit_open.o giounit_util.o
! MDOBJ = mdunistd.o paraenv.o
  #
  # -DINCONSISTENT_THREAD_COMP
  #    when thread computing is inconsistent, define this
! DEFINES = -O3 ${GDEFS} -DINCONSISTENT_THREAD_COMP
  
! FFLAGS = -O3 -fopenmp -msse3 -fno-inline-functions ${DEFINES} -I/opt/local/include
! F77FLAGS =
! F90FLAGS = -ffree-line-length-none
! FOPTNON = -O0
! 
! LDFLAGS = -fopenmp
! CFLAGS = -fopenmp -O3 -fno-inline-functions -I/opt/local/include
! CXXFLAGS = -fopenmp -O3 -fno-inline-functions -I/opt/local/include
! 
! LIBLAPACK = -L/opt/local/lib -lfftw3_threads -lfftw3 -llapack
! LIBBLAS = -lblas
! EXTRALIBS = -lstdc++
! 
! XMPI_OBJ = xmpi_univ.o
! EIGSYSTM_OBJ = eigsystm.o
! EIGSYSTM_OBJ = eigsystm-lapack.o
! #EIGSYSTM_OBJ = eigsystm-scalapack.o
! FFT3D_OBJ = fft3d_fftw.o
! FFTGRID_H_SRC = fftgrid-scl-fftw.h
  
! CC = gcc
! CXX = g++
! FC_CIF = gfortran
! FFLAG_CIF = 
  #--------------------------------
  # for K computer
  #
***************
*** 535,542 ****
  	${CPP} -P ${DEFINES} $< | sed '/^ *$$/d' > $*.f
  
  install: default
! 	install -d $(DESTDIR)/usr/bin
! 	install inipot cgmrpt mdrpt vbpef vbstm wfn2chg wannier pefcos xticonv strconv diffstr iplstr cmpstr pe2dos tetrapdos hmatr2bnd atmcdist distpart $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 535,542 ----
  	${CPP} -P ${DEFINES} $< | sed '/^ *$$/d' > $*.f
  
  install: default
! 	mkdir -p $(PREFIX)/bin
! 	cp -p inipot cgmrpt mdrpt vbpef vbstm wfn2chg wannier pefcos xticonv strconv diffstr iplstr cmpstr pe2dos tetrapdos hmatr2bnd atmcdist distpart $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp_20180420.orig/xtapp-util_20180420/fldtool/Makefile xtapp_20180420/xtapp-util_20180420/fldtool/Makefile
*** xtapp_20180420.orig/xtapp-util_20180420/fldtool/Makefile	Thu Feb 14 16:02:12 2019
--- xtapp_20180420/xtapp-util_20180420/fldtool/Makefile	Thu Feb 14 16:15:34 2019
***************
*** 93,104 ****
  	fftgrid.h
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install \
  	wfn2dx wfk2dx lpt2dx rho2dx rok2dx stm2dx rhomm lptmm wfn2ee \
  	wfn2respack rho2chgcar \
  	exrho2dx stm2cci cmprho cmpband wfneig rho2xsf cmpamat cmppdos \
! 	$(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 93,104 ----
  	fftgrid.h
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p \
  	wfn2dx wfk2dx lpt2dx rho2dx rok2dx stm2dx rhomm lptmm wfn2ee \
  	wfn2respack rho2chgcar \
  	exrho2dx stm2cci cmprho cmpband wfneig rho2xsf cmpamat cmppdos \
! 	$(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp_20180420.orig/xtapp-util_20180420/inputconv/Makefile xtapp_20180420/xtapp-util_20180420/inputconv/Makefile
*** xtapp_20180420.orig/xtapp-util_20180420/inputconv/Makefile	Thu Feb 14 16:02:12 2019
--- xtapp_20180420/xtapp-util_20180420/inputconv/Makefile	Thu Feb 14 16:15:49 2019
***************
*** 39,46 ****
  	inputconv a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install inputconv $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 39,46 ----
  	inputconv a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p inputconv $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp_20180420.orig/xtapp-util_20180420/mdtool/Makefile xtapp_20180420/xtapp-util_20180420/mdtool/Makefile
*** xtapp_20180420.orig/xtapp-util_20180420/mdtool/Makefile	Thu Feb 14 16:02:12 2019
--- xtapp_20180420/xtapp-util_20180420/mdtool/Makefile	Thu Feb 14 16:13:38 2019
***************
*** 40,47 ****
  	scanmdlog tailmdlog a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install scanmdlog tailmdlog $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 40,47 ----
  	scanmdlog tailmdlog a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p scanmdlog tailmdlog $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp_20180420.orig/xtapp-util_20180420/minmaxband/Makefile xtapp_20180420/xtapp-util_20180420/minmaxband/Makefile
*** xtapp_20180420.orig/xtapp-util_20180420/minmaxband/Makefile	Thu Feb 14 16:02:12 2019
--- xtapp_20180420/xtapp-util_20180420/minmaxband/Makefile	Thu Feb 14 16:16:16 2019
***************
*** 22,29 ****
  	minmaxband a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install minmaxband $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 22,29 ----
  	minmaxband a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p minmaxband $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
diff -crN xtapp_20180420.orig/xtapp-util_20180420/wfchg2pdos/Makefile xtapp_20180420/xtapp-util_20180420/wfchg2pdos/Makefile
*** xtapp_20180420.orig/xtapp-util_20180420/wfchg2pdos/Makefile	Thu Feb 14 16:02:12 2019
--- xtapp_20180420/xtapp-util_20180420/wfchg2pdos/Makefile	Thu Feb 14 16:15:06 2019
***************
*** 40,47 ****
  	ltzpdos wfchg2pdos a.out
  
  install: tools
! 	install -d $(DESTDIR)/usr/bin
! 	install ltzpdos wfchg2pdos $(DESTDIR)/usr/bin
  
  # Local Variables: ***
  # mode:Makefile ***
--- 40,47 ----
  	ltzpdos wfchg2pdos a.out
  
  install: tools
! 	mkdir -p $(PREFIX)/bin
! 	cp -p ltzpdos wfchg2pdos $(PREFIX)/bin
  
  # Local Variables: ***
  # mode:Makefile ***
