diff -crN scalapack-2.0.2.orig/BLACS/SRC/cgamn2d_.c scalapack-2.0.2/BLACS/SRC/cgamn2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/cgamn2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/cgamn2d_.c	Thu Aug  1 17:41:27 2019
***************
*** 221,227 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 221,227 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
diff -crN scalapack-2.0.2.orig/BLACS/SRC/cgamx2d_.c scalapack-2.0.2/BLACS/SRC/cgamx2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/cgamx2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/cgamx2d_.c	Thu Aug  1 17:41:37 2019
***************
*** 221,227 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 221,227 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
diff -crN scalapack-2.0.2.orig/BLACS/SRC/dgamn2d_.c scalapack-2.0.2/BLACS/SRC/dgamn2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/dgamn2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/dgamn2d_.c	Thu Aug  1 17:41:47 2019
***************
*** 221,227 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 221,227 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
diff -crN scalapack-2.0.2.orig/BLACS/SRC/dgamx2d_.c scalapack-2.0.2/BLACS/SRC/dgamx2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/dgamx2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/dgamx2d_.c	Thu Aug  1 17:41:52 2019
***************
*** 221,227 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 221,227 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
diff -crN scalapack-2.0.2.orig/BLACS/SRC/igamn2d_.c scalapack-2.0.2/BLACS/SRC/igamn2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/igamn2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/igamn2d_.c	Thu Aug  1 17:41:59 2019
***************
*** 218,224 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 218,224 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
diff -crN scalapack-2.0.2.orig/BLACS/SRC/igamx2d_.c scalapack-2.0.2/BLACS/SRC/igamx2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/igamx2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/igamx2d_.c	Thu Aug  1 17:42:05 2019
***************
*** 218,224 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 218,224 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
diff -crN scalapack-2.0.2.orig/BLACS/SRC/sgamn2d_.c scalapack-2.0.2/BLACS/SRC/sgamn2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/sgamn2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/sgamn2d_.c	Thu Aug  1 17:42:14 2019
***************
*** 221,227 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 221,227 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
diff -crN scalapack-2.0.2.orig/BLACS/SRC/sgamx2d_.c scalapack-2.0.2/BLACS/SRC/sgamx2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/sgamx2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/sgamx2d_.c	Thu Aug  1 17:42:19 2019
***************
*** 221,227 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 221,227 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
diff -crN scalapack-2.0.2.orig/BLACS/SRC/zgamn2d_.c scalapack-2.0.2/BLACS/SRC/zgamn2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/zgamn2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/zgamn2d_.c	Thu Aug  1 17:42:26 2019
***************
*** 221,227 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 221,227 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
diff -crN scalapack-2.0.2.orig/BLACS/SRC/zgamx2d_.c scalapack-2.0.2/BLACS/SRC/zgamx2d_.c
*** scalapack-2.0.2.orig/BLACS/SRC/zgamx2d_.c	Wed Apr 25 01:26:51 2012
--- scalapack-2.0.2/BLACS/SRC/zgamx2d_.c	Thu Aug  1 17:42:31 2019
***************
*** 221,227 ****
        {
  #endif
        i = 2;
!       ierr=MPI_Type_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
--- 221,227 ----
        {
  #endif
        i = 2;
!       ierr=MPI_Type_create_struct(i, len, disp, dtypes, &MyType);
        ierr=MPI_Type_commit(&MyType);
        bp->N = bp2->N = 1;
        bp->dtype = bp2->dtype = MyType;
