--- ./install/make.inc.in.orig	2016-09-23 16:00:53.499397534 +0200
+++ ./install/make.inc.in	2016-09-23 16:01:09.635396988 +0200
@@ -41,7 +41,7 @@
 
 # MANUAL_DFLAGS  = additional precompilation option(s), if desired
 #                  BEWARE: it does not work for IBM xlf! Manually edit FDFLAGS
-MANUAL_DFLAGS  =
+MANUAL_DFLAGS  = -D__ELPA_2016
 DFLAGS         = @dflags@
 FDFLAGS        = @fdflags@
 
