diff -crN lammps-20190807.orig/cmake/CMakeLists.txt lammps-20190807/cmake/CMakeLists.txt
*** lammps-20190807.orig/cmake/CMakeLists.txt	Wed Aug  7 00:17:40 2019
--- lammps-20190807/cmake/CMakeLists.txt	Sun Sep  8 15:00:49 2019
***************
*** 26,31 ****
--- 26,44 ----
  file(GLOB LMP_SOURCES ${LAMMPS_SOURCE_DIR}/main.cpp)
  list(REMOVE_ITEM LIB_SOURCES ${LMP_SOURCES})
  
+ # RPATH fix
+ set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+   set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+ else()
+   set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+ endif()
+ 
+ #policy update CMP0042
+ if(APPLE)
+   set(CMAKE_MACOSX_RPATH ON)
+ endif()
+ 
  # Cmake modules/macros are in a subdirectory to keep this file cleaner
  set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Modules)
  
***************
*** 83,96 ****
  ########################################################################
  # User input options                                                   #
  ########################################################################
! option(BUILD_EXE "Build lmp binary" ON)
  if(BUILD_EXE)
!   set(LAMMPS_MACHINE "" CACHE STRING "Suffix to append to lmp binary (WON'T enable any features automatically")
    mark_as_advanced(LAMMPS_MACHINE)
    if(LAMMPS_MACHINE)
      set(LAMMPS_MACHINE "_${LAMMPS_MACHINE}")
    endif()
!   set(LAMMPS_BINARY lmp${LAMMPS_MACHINE})
  endif()
  
  option(BUILD_LIB "Build LAMMPS library" OFF)
--- 96,109 ----
  ########################################################################
  # User input options                                                   #
  ########################################################################
! option(BUILD_EXE "Build lammps binary" ON)
  if(BUILD_EXE)
!   set(LAMMPS_MACHINE "" CACHE STRING "Suffix to append to lammps binary (WON'T enable any features automatically")
    mark_as_advanced(LAMMPS_MACHINE)
    if(LAMMPS_MACHINE)
      set(LAMMPS_MACHINE "_${LAMMPS_MACHINE}")
    endif()
!   set(LAMMPS_BINARY lammps${LAMMPS_MACHINE})
  endif()
  
  option(BUILD_LIB "Build LAMMPS library" OFF)
diff -crN lammps-20190807.orig/cmake/Modules/Packages/MESSAGE.cmake lammps-20190807/cmake/Modules/Packages/MESSAGE.cmake
*** lammps-20190807.orig/cmake/Modules/Packages/MESSAGE.cmake	Wed Aug  7 00:17:40 2019
--- lammps-20190807/cmake/Modules/Packages/MESSAGE.cmake	Sun Sep  8 14:50:35 2019
***************
*** 21,27 ****
      target_link_libraries(cslib PUBLIC ${ZMQ_LIBRARIES})
    else()
      target_compile_definitions(cslib PRIVATE -DZMQ_NO)
!     target_include_directories(cslib PRIVATE ${LAMMPS_LIB_SOURCE_DIR}/message/cslib/src/STUBS_ZMQ)
    endif()
  
    list(APPEND LAMMPS_LINK_LIBS cslib)
--- 21,27 ----
      target_link_libraries(cslib PUBLIC ${ZMQ_LIBRARIES})
    else()
      target_compile_definitions(cslib PRIVATE -DZMQ_NO)
!     target_include_directories(cslib BEFORE PRIVATE ${LAMMPS_LIB_SOURCE_DIR}/message/cslib/src/STUBS_ZMQ)
    endif()
  
    list(APPEND LAMMPS_LINK_LIBS cslib)
diff -crN lammps-20190807.orig/src/memory.cpp lammps-20190807/src/memory.cpp
*** lammps-20190807.orig/src/memory.cpp	Wed Aug  7 00:17:40 2019
--- lammps-20190807/src/memory.cpp	Mon Sep  9 17:38:52 2019
***************
*** 13,18 ****
--- 13,19 ----
  
  #include "memory.h"
  #include <cstdlib>
+ #include <string.h>
  #include "error.h"
  
  #if defined(LMP_USER_INTEL) && defined(__INTEL_COMPILER)
