diff -urN ermod-0.2.4.orig/engproc.F90 ermod-0.2.4/engproc.F90
--- ermod-0.2.4.orig/engproc.F90	2013-02-12 23:52:06.000000000 +0900
+++ ermod-0.2.4/engproc.F90	2015-06-08 09:10:16.696010634 +0900
@@ -321,7 +321,7 @@
     logical :: skipcond
     logical, save :: pme_initialized = .false.
     ! FIXME: this call to mpi_info is sprious
-    call mpi_info                                                    ! MPI
+    call mpi_set_info                                                    ! MPI
     call mpi_init_active_group(nactiveproc)
 
     call sanity_check_sluvid()
@@ -453,7 +453,7 @@
     real, parameter :: tiny=1.0e-30
     real, dimension(:), allocatable :: sve1,sve2
     character(len=11) :: formtype="UNFORMATTED"
-    call mpi_info                                                    ! MPI
+    call mpi_set_info                                                    ! MPI
     !
 
     ! synchronize voffset
diff -urN ermod-0.2.4.orig/mpiproc.F90 ermod-0.2.4/mpiproc.F90
--- ermod-0.2.4.orig/mpiproc.F90	2013-02-10 17:17:46.000000000 +0900
+++ ermod-0.2.4/mpiproc.F90	2015-06-08 09:11:17.728677718 +0900
@@ -59,7 +59,7 @@
 #ifdef PERF
        walltime = MPI_WTIME()
 #endif
-       call mpi_info
+       call mpi_set_info
        times(:) = 0.0
     endif
     if(type.eq.'stop') then                                          ! MPI
@@ -79,7 +79,7 @@
 #endif
   end subroutine mpi_setup                                                   ! MPI
 
-  subroutine mpi_info                                              ! MPI
+  subroutine mpi_set_info                                              ! MPI
     nprocs=1
     myrank=0
 #ifndef noMPI
@@ -87,7 +87,7 @@
     call mpi_comm_rank(mpi_comm_world,myrank,ierror)                 ! MPI
 #endif
     return
-  end subroutine mpi_info                                                   ! MPI
+  end subroutine mpi_set_info                                                   ! MPI
 
   subroutine mpi_abend()
     integer :: ierror
diff -urN ermod-0.2.4.orig/setconf.F90 ermod-0.2.4/setconf.F90
--- ermod-0.2.4.orig/setconf.F90	2013-02-12 23:52:06.000000000 +0900
+++ ermod-0.2.4/setconf.F90	2015-06-08 09:09:39.795816577 +0900
@@ -195,7 +195,7 @@
     real, parameter :: tiny=1.0e-20
     real :: real_seed
     character*3 scrtype
-    call mpi_info                                                    ! MPI
+    call mpi_set_info                                                    ! MPI
 
     intprm=1                    ! trajectory reading
     if(intprm.eq.1) then        ! default settings for trajectory reading
