diff -crN lapack-3.7.0.orig/SRC/Makefile lapack-3.7.0/SRC/Makefile
*** lapack-3.7.0.orig/SRC/Makefile	2016-12-24 08:01:32.000000000 +0900
--- lapack-3.7.0/SRC/Makefile	2017-02-20 17:18:41.000000000 +0900
***************
*** 464,470 ****
     zgelqt.o zgelqt3.o zgemlqt.o \
     zgetsls.o zgeqr.o zlatsqr.o zlamtsqr.o zgemqr.o \
     zgelq.o zlaswlq.o zlamswlq.o zgemlq.o \
-    ztplqt.o ztplqt2.o ztpmlqt.o \
     zhetrd_2stage.o zhetrd_he2hb.o zhetrd_hb2st.o zhb2st_kernels.o \
     zheevd_2stage.o zheev_2stage.o zheevx_2stage.o zheevr_2stage.o \
     zhbev_2stage.o zhbevx_2stage.o zhbevd_2stage.o zhegv_2stage.o
--- 464,469 ----
***************
*** 558,564 ****
  
  .F.o:
  	$(FORTRAN) $(OPTS) -c $< -o $@
! 	
  slaruv.o: slaruv.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
  dlaruv.o: dlaruv.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
  sla_wwaddw.o: sla_wwaddw.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
--- 557,563 ----
  
  .F.o:
  	$(FORTRAN) $(OPTS) -c $< -o $@
! 
  slaruv.o: slaruv.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
  dlaruv.o: dlaruv.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
  sla_wwaddw.o: sla_wwaddw.f ; $(FORTRAN) $(NOOPT) -c -o $@ $<
