diff -urN q-e-qe-7.2.orig/CMakeLists.txt q-e-qe-7.2/CMakeLists.txt
--- q-e-qe-7.2.orig/CMakeLists.txt	2023-03-28 01:12:09
+++ q-e-qe-7.2/CMakeLists.txt	2024-02-17 14:00:53
@@ -500,16 +500,8 @@
 ###########################################################
 # SCALAPACK
 # The following targets will be defined:
-add_library(qe_scalapack INTERFACE)
-qe_install_targets(qe_scalapack)
-###########################################################
 if(QE_ENABLE_SCALAPACK)
-    find_package(SCALAPACK REQUIRED QUIET)
-    message(STATUS "Found SCALAPACK: ${SCALAPACK_LIBRARIES};${SCALAPACK_LINKER_FLAGS}")
-    target_link_libraries(qe_scalapack
-        INTERFACE
-            ${SCALAPACK_LIBRARIES}
-            ${SCALAPACK_LINKER_FLAGS})
+    find_package(scalapack REQUIRED)
     if(QE_ENABLE_SCALAPACK_QRCP)
         include(CheckFortranFunctionExists)
         set(CMAKE_REQUIRED_LIBRARIES "${SCALAPACK_LIBRARIES}")
@@ -552,7 +544,7 @@
             ${ELPA_LIBRARIES}
             ${ELPA_LIBRARIES_DEP}
             ${ELPA_LINKER_FLAGS}
-            qe_scalapack)
+            scalapack)
     target_include_directories(qe_elpa
         INTERFACE
             ${ELPA_Fortran_MODS_DIR}
diff -urN q-e-qe-7.2.orig/GWW/CMakeLists.txt q-e-qe-7.2/GWW/CMakeLists.txt
--- q-e-qe-7.2.orig/GWW/CMakeLists.txt	2023-03-28 01:12:09
+++ q-e-qe-7.2/GWW/CMakeLists.txt	2024-02-17 14:00:53
@@ -183,7 +183,7 @@
 qe_add_library(qe_gww_pw4gww ${src_pw4gww})
 target_link_libraries(qe_gww_pw4gww
     PRIVATE
-        qe_scalapack
+        scalapack
         qe_pw
         qe_modules
         qe_gww
diff -urN q-e-qe-7.2.orig/KS_Solvers/CMakeLists.txt q-e-qe-7.2/KS_Solvers/CMakeLists.txt
--- q-e-qe-7.2.orig/KS_Solvers/CMakeLists.txt	2023-03-28 01:12:09
+++ q-e-qe-7.2/KS_Solvers/CMakeLists.txt	2024-02-17 14:00:53
@@ -109,7 +109,7 @@
 qe_add_library(qe_kssolver_ppcg ${src_ppcg})
 target_link_libraries(qe_kssolver_ppcg
     PRIVATE
-        qe_scalapack
+        scalapack
         qe_openmp_fortran
         qe_lax
         qe_utilx
diff -urN q-e-qe-7.2.orig/LAXlib/CMakeLists.txt q-e-qe-7.2/LAXlib/CMakeLists.txt
--- q-e-qe-7.2.orig/LAXlib/CMakeLists.txt	2023-03-28 01:12:09
+++ q-e-qe-7.2/LAXlib/CMakeLists.txt	2024-02-17 14:00:53
@@ -18,7 +18,7 @@
     PRIVATE
         qe_elpa
         qe_lapack
-        qe_scalapack
+        scalapack
         qe_openmp_fortran
         qe_mpi_fortran
         qe_utilx
diff -urN q-e-qe-7.2.orig/PW/CMakeLists.txt q-e-qe-7.2/PW/CMakeLists.txt
--- q-e-qe-7.2.orig/PW/CMakeLists.txt	2023-03-28 01:12:09
+++ q-e-qe-7.2/PW/CMakeLists.txt	2024-02-17 14:00:53
@@ -303,7 +303,7 @@
         qe_device_lapack  
         qe_hdf5_fortran
         qe_elpa
-        qe_scalapack
+        scalapack
         qe_lax
         qe_kssolver_davidson
         qe_kssolver_cg
